File: basaltpass-backend/internal/dto/team/team.go
package team

import "basaltpass-backend/internal/model"

// ListTeamsRequest 团队列表查询参数
type ListTeamsRequest struct {
	Page    int    `query:"page"`
	Limit   int    `query:"limit"`
	Keyword string `query:"keyword"`
}

type TeamBrief struct {
	ID          uint   `json:"id"`
	Name        string `json:"name"`
	Description string `json:"description"`
	AvatarURL   string `json:"avatar_url"`
	IsActive    bool   `json:"is_active"`
	MemberCount int    `json:"member_count"`
	CreatedAt   string `json:"created_at"`
}

type ListTeamsResponse struct {
	Teams []TeamBrief `json:"teams"`
	Total int         `json:"total"`
}

type AdminUpdateTeamRequest struct {
	Name        string `json:"name"`
	Description string `json:"description"`
	AvatarURL   string `json:"avatar_url"`
	IsActive    *bool  `json:"is_active"`
}

// AdminCreateTeamRequest 创建团队
type AdminCreateTeamRequest struct {
	Name        string `json:"name"`
	Description string `json:"description"`
	AvatarURL   string `json:"avatar_url"`
	OwnerUserID uint   `json:"owner_user_id"`
}

type TeamMemberResponse struct {
	ID        uint   `json:"id"`
	TeamID    uint   `json:"team_id"`
	UserID    uint   `json:"user_id"`
	Role      string `json:"role"`
	Status    string `json:"status"`
	JoinedAt  string `json:"joined_at"`
	CreatedAt string `json:"created_at"`
	User      struct {
		ID        uint   `json:"id"`
		Email     string `json:"email"`
		Nickname  string `json:"nickname"`
		AvatarURL string `json:"avatar_url"`
	} `json:"user"`
}

type AddMemberRequest struct {
	UserID uint           `json:"user_id"`
	Role   model.TeamRole `json:"role"`
}

File: basaltpass-frontend/src/shared/api/user/team.ts
import client from '../client';

export interface CreateTeamRequest {
  name: string;
  description?: string;
  avatar_url?: string;
}

export interface UpdateTeamRequest {
  name?: string;
  description?: string;
  avatar_url?: string;
}

export interface TeamResponse {
  id: number;
  name: string;
  description: string;
  avatar_url: string;
  is_active: boolean;
  created_at: string;
  updated_at: string;
  member_count: number;
  user_role?: string;
}

export interface TeamMemberResponse {
  id: number;
  team_id: number;
  user_id: number;
  role: string;
  status: string;
  joined_at: string;
  created_at: string;
  user: {
    id: number;
    email: string;
    nickname: string;
    avatar_url: string;
  };
}

export interface AddMemberRequest {
  user_id: number;
  role: 'owner' | 'admin' | 'member';
}

export interface UpdateMemberRoleRequest {
  role: 'owner' | 'admin' | 'member';
}

export interface UserTeamResponse {
  team_id: number;
  team_name: string;
  role: string;
  joined_at: string;
}

// teamAPI
export const teamApi = {
  // createteam
  createTeam: (data: CreateTeamRequest) =>
    client.post<{ data: { team: TeamResponse }; message: string }>('/api/v1/teams', data),

  // getusertranslatedhasteam
  getUserTeams: () =>
    client.get<{ data: UserTeamResponse[] }>('/api/v1/teams'),

  // getteaminfo
  getTeam: (id: number) =>
    client.get<{ data: TeamResponse }>(`/api/v1/teams/${id}`),

  // updateteaminfo
  updateTeam: (id: number, data: UpdateTeamRequest) =>
    client.put<{ message: string }>(`/api/v1/teams/${id}`, data),

  // deleteteam
  deleteTeam: (id: number) =>
    client.delete<{ message: string }>(`/api/v1/teams/${id}`),

  // getteamtranslatedlist
  getTeamMembers: (id: number) =>
    client.get<{ data: TeamMemberResponse[] }>(`/api/v1/teams/${id}/members`),

  // translatedteamtranslated
  addMember: (id: number, data: AddMemberRequest) =>
    client.post<{ message: string }>(`/api/v1/teams/${id}/members`, data),

  // updatetranslatedrole
  updateMemberRole: (teamId: number, memberId: number, data: UpdateMemberRoleRequest) =>
    client.put<{ message: string }>(`/api/v1/teams/${teamId}/members/${memberId}`, data),

  // translatedteamtranslated
  removeMember: (teamId: number, memberId: number) =>
    client.delete<{ message: string }>(`/api/v1/teams/${teamId}/members/${memberId}`),

  // translatedteam
  leaveTeam: (id: number) =>
    client.post<{ message: string }>(`/api/v1/teams/${id}/leave`),
}; 